File: Support/Templates/content/CslaLibrary-CSharp/DynamicRootBindingList.cs
using System;
using System.Collections.Generic;
using Csla;

namespace Company.CslaLibrary1
{
  [Serializable]
  public class DynamicRootBindingList :
    DynamicBindingListBase<DynamicRoot>
  {
    protected override object AddNewCore()
    {
      var dataPortal = ApplicationContext.GetRequiredService<IDataPortal<DynamicRoot>>();
      DynamicRoot item = dataPortal.Create();
      Add(item);
      return item;
    }

    private static void AddObjectAuthorizationRules()
    {
      // TODO: add authorization rules
      //AuthorizationRules.AllowGet(typeof(DynamicRootBindingList), "Role");
      //AuthorizationRules.AllowEdit(typeof(DynamicRootBindingList), "Role");
    }

    [Fetch]
    private void Fetch()
    {
      // TODO: load values
      RaiseListChangedEvents = false;
      object listData = null;
      var dataPortal = ApplicationContext.GetRequiredService<IDataPortal<DynamicRoot>>();
      foreach (var item in (List<object>)listData)
        Add(dataPortal.Fetch(item));
      RaiseListChangedEvents = true;
    }
  }
}

File: Support/Templates/content/CslaLibrary-VisualBasic/DynamicRootBindingList.vb
Imports Csla
Imports Csla.Security

<Serializable()>
Public Class DynamicRootBindingList
  Inherits DynamicBindingListBase(Of DynamicRoot)

  Protected Overrides Function AddNewCore() As Object

    Dim dataPortal As IDataPortal(Of DynamicRoot) = ApplicationContext.GetRequiredService(Of IDataPortal(Of DynamicRoot))()
    Dim item As DynamicRoot = dataPortal.Create()
    Add(item)
    Return item

  End Function

  Private Shared Sub AddObjectAuthorizationRules()
    'TODO: add authorization rules
    'AuthorizationRules.AllowGet(GetType(DynamicRootBindingList), "Role")
    'AuthorizationRules.AllowEdit(GetType(DynamicRootBindingList), "Role")
  End Sub

  <Fetch>
  Private Sub Fetch()

    ' TODO: load values
    RaiseListChangedEvents = False
    Dim listData As Object = Nothing
    Dim dataPortal As IDataPortal(Of DynamicRoot) = ApplicationContext.GetRequiredService(Of IDataPortal(Of DynamicRoot))()
    For Each oneItemData As Object In CType(listData, List(Of Object))
      Add(dataPortal.Fetch(oneItemData))
    Next
    RaiseListChangedEvents = True

  End Sub

End Class
